### DISCUSSION TITLE
Row ID Case Not Found in UUID[] Array

### DISCUSSION THREAD

**Discussion by Closedd99** [supabase/supabase#30677] [Questions] [answered]
Been stuck with this roadblock for more than 24 hours now, so any help would be greatly appreciated!

 Looking to create a trigger function that updates a field on multiple rows with uuid's found inside a uuid array, comes back with an error "case not found"

Tables involved: 
"receiptsTb" (table)
"replenishmentsTb" (table)
replensubview (table view)

Fields involved:
"replenishmentsTb".child_receipts_id type as uuid array
replensubview.child_receipts_id type as uuid  array
"receiptsTb".receipt_status type as text

Progression:
1. Before row insert trigger function in "replenishmentsTb", NEW.child_receipt_id is set from replensubview.child_receipts_id (uuid array)-- no issues here
2. Row is inserted -- no issues here
3. After row insert trigger function, updating "receiptsTb".receipt_status = "Outstanding" where "receiptsTb".id  is found in (replenishmentsTb).NEW.child_receipt_id uuid array -- issue: all i get is "case not found" error on the error logs.

the exact snippet I used to look for the relevant receiptsTb.id is:
`"WHERE id IN (Select unnest(NEW.child_receipts_id::uuid[])::uuid)"`and other variations, including using ANY and both with or without casting uuid[] or uuid for NEW.child_receipts_id.

I am 100% sure and double/triple checked that all the UUID inside the child_receipts_id uuid array are all existing and correct
What am I missing here? Is there a better approach to updating multiple rows from  a uuid array? Thank you in advance!

PS: sample child_receipts_id uuid array: 
```
[
  "cebf1642-4587-4175-8053-8d7e7b656ede",
  "15bca170-7347-46c2-8cf1-c2e94cf58bf2",
  "c195a720-4dd2-47bb-ba23-f415f5f361fc"
]
```

**Comment by GaryAustin1** [comment]
In Discord you showed the function and there was no Case statement in it.  I believe that error is about a Case statement.  Do you have another function that might be called in this process?

**Reply by GaryAustin1** [reply, marked as answer]
From Discord this turned out to be another trigger function being called with a missing ELSE clause on the CASE statement in that,

**Reply by Closedd99** [reply]
yes, was definitely an oversight from my part. got it working already! thanks for the help
